girder/api/v1/group.py

```diff
@@ -55,25 +55,41 @@ def __init__(self):
     @filtermodel(model=GroupModel)
     @autoDescribeRoute(
         Description('Search for groups or list all groups.')
-        .param('text', 'Pass this to perform a full-text search for groups.', required=False)
+        .notes('If you pass a "query" field, you can also pass'
+               'a "mode" field to select a different search mode. By default a full-text search'
+               'is provided.')
+        .param('query', 'Pass this to perform a search for groups.', required=False)
+        .param('mode', 'Pass to search with a different search mode.', required=False)
         .param('exact', 'If true, only return exact name matches. This is '
                'case sensitive.', required=False, dataType='boolean', default=False)
         .pagingParams(defaultSort='name')
         .errorResponse()
     )
-    def find(self, text, exact, limit, offset, sort):
+    def find(self, query, mode, exact, limit, offset, sort):
         user = self.getCurrentUser()
-        if text is not None:
+
```

Comment:
May be relevant to add a search helper function, the following block seems to be repeated over and over